async-query-core/src/main/java/org/opensearch/sql/spark/parameter/SparkParameterComposerCollection.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.parameter;
+
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import org.opensearch.sql.datasource.model.DataSourceMetadata;
+import org.opensearch.sql.datasource.model.DataSourceType;
+import org.opensearch.sql.spark.asyncquery.model.AsyncQueryRequestContext;
+import org.opensearch.sql.spark.dispatcher.model.DispatchQueryRequest;
+
+/** Stores Spark parameter composers and dispatch compose request to each composer */
+public class SparkParameterComposerCollection {
+  Collection<GeneralSparkParameterComposer> generalComposers = new ArrayList<>();
+  Map<DataSourceType, Collection<DataSourceSparkParameterComposer>> datasourceComposers =
```

Comment:
why need map? how about Collection<DataSourceSparkParameterComposer>?

Reply:
We want to lookup by DataSourceType. Do you mean we want to lookup using linear search?

Reply:
no, i mean why not use Collection of DataSourceSparkParameterComposer.
understood now,